unit-tests-java8/src/test/java/org/eclipse/collections/test/bag/mutable/sorted/MutableBagIterableTestCase.java

```diff
@@ -37,6 +38,17 @@ default void MutableBagIterable_addOccurrences()
         assertThrows(
                 IllegalArgumentException.class,
                 () -> mutableBag.addOccurrences(4, -1));
+        assertEquals(Bags.immutable.with(1, 1, 1, 2, 2, 3, 3, 3, 4, 4, 4, 4), mutableBag);
+        assertEquals(Bags.immutable.with(1, 1, 1, 2, 2, 3, 3, 3, 4, 4, 4, 4), mutableBag);
+
+        assertThrows(
+                ArithmeticException.class,
+                () -> mutableBag.addOccurrences(3, Integer.MAX_VALUE - 1));
+        assertEquals(Bags.immutable.with(1, 1, 1, 2, 2, 3, 3, 3, 4, 4, 4, 4), mutableBag);
+
+        mutableBag.addOccurrences(5, Integer.MAX_VALUE - 10);
+        mutableBag.addOccurrences(6, Integer.MAX_VALUE - 10);
+        Assert.assertEquals(-10, mutableBag.size());
```

Comment:
But we can still overflow size with big numbers of occurrences at several keys.

Reply:
Java collection's `size` function has been known to have integer limits for a very long time. And for almost all applications, it doesn't matter. If some application needs a specialty collection with larger sizes, they should pick a suitable implementation, no?